--- src/interface.c	2004-06-27 23:21:02.000000000 +0900
+++ src/interface-libgtk2.0.c	2004-07-05 20:50:45.000000000 +0900
@@ -156,8 +156,7 @@
   GtkWidget *about1;
   GtkWidget *image2673;
   GtkWidget *toolbar;
-  gint tmp_toolbar_icon_size;
-  GtkWidget *tmp_image;
+  GtkWidget *tmp_toolbar_icon;
   GtkWidget *import_button;
   GtkWidget *add_files_button;
   GtkWidget *add_dirs_button;
@@ -697,55 +696,75 @@
   gtk_widget_show (toolbar);
   gtk_box_pack_start (GTK_BOX (vbox1), toolbar, FALSE, FALSE, 0);
   gtk_toolbar_set_style (GTK_TOOLBAR (toolbar), GTK_TOOLBAR_BOTH);
-  tmp_toolbar_icon_size = gtk_toolbar_get_icon_size (GTK_TOOLBAR (toolbar));
 
-  tmp_image = gtk_image_new_from_stock ("gtk-goto-top", tmp_toolbar_icon_size);
-  gtk_widget_show (tmp_image);
-  import_button = (GtkWidget*) gtk_tool_button_new (tmp_image, _("Read"));
+  tmp_toolbar_icon = gtk_image_new_from_stock ("gtk-goto-top", gtk_toolbar_get_icon_size (GTK_TOOLBAR (toolbar)));
+  import_button = gtk_toolbar_append_element (GTK_TOOLBAR (toolbar),
+                                GTK_TOOLBAR_CHILD_BUTTON,
+                                NULL,
+                                _("Read"),
+                                _("Read Existing Database from Disk/iPod"), NULL,
+                                tmp_toolbar_icon, NULL, NULL);
+  gtk_label_set_use_underline (GTK_LABEL (((GtkToolbarChild*) (g_list_last (GTK_TOOLBAR (toolbar)->children)->data))->label), TRUE);
   gtk_widget_show (import_button);
-  gtk_container_add (GTK_CONTAINER (toolbar), import_button);
-  gtk_tool_item_set_tooltip (GTK_TOOL_ITEM (import_button), tooltips, _("Read Existing Database from Disk/iPod"), NULL);
 
-  tmp_image = gtk_image_new_from_stock ("gtk-add", tmp_toolbar_icon_size);
-  gtk_widget_show (tmp_image);
-  add_files_button = (GtkWidget*) gtk_tool_button_new (tmp_image, _("Files"));
+  tmp_toolbar_icon = gtk_image_new_from_stock ("gtk-add", gtk_toolbar_get_icon_size (GTK_TOOLBAR (toolbar)));
+  add_files_button = gtk_toolbar_append_element (GTK_TOOLBAR (toolbar),
+                                GTK_TOOLBAR_CHILD_BUTTON,
+                                NULL,
+                                _("Files"),
+                                _("Add Files or Directories"), NULL,
+                                tmp_toolbar_icon, NULL, NULL);
+  gtk_label_set_use_underline (GTK_LABEL (((GtkToolbarChild*) (g_list_last (GTK_TOOLBAR (toolbar)->children)->data))->label), TRUE);
   gtk_widget_show (add_files_button);
-  gtk_container_add (GTK_CONTAINER (toolbar), add_files_button);
-  gtk_tool_item_set_tooltip (GTK_TOOL_ITEM (add_files_button), tooltips, _("Add Files or Directories"), NULL);
 
-  tmp_image = gtk_image_new_from_stock ("gtk-add", tmp_toolbar_icon_size);
-  gtk_widget_show (tmp_image);
-  add_dirs_button = (GtkWidget*) gtk_tool_button_new (tmp_image, _("Dirs"));
+  tmp_toolbar_icon = gtk_image_new_from_stock ("gtk-add", gtk_toolbar_get_icon_size (GTK_TOOLBAR (toolbar)));
+  add_dirs_button = gtk_toolbar_append_element (GTK_TOOLBAR (toolbar),
+                                GTK_TOOLBAR_CHILD_BUTTON,
+                                NULL,
+                                _("Dirs"),
+                                _("Add Directories Recursively"), NULL,
+                                tmp_toolbar_icon, NULL, NULL);
+  gtk_label_set_use_underline (GTK_LABEL (((GtkToolbarChild*) (g_list_last (GTK_TOOLBAR (toolbar)->children)->data))->label), TRUE);
   gtk_widget_show (add_dirs_button);
-  gtk_container_add (GTK_CONTAINER (toolbar), add_dirs_button);
-  gtk_tool_item_set_tooltip (GTK_TOOL_ITEM (add_dirs_button), tooltips, _("Add Directories Recursively"), NULL);
 
-  tmp_image = gtk_image_new_from_stock ("gtk-add", tmp_toolbar_icon_size);
-  gtk_widget_show (tmp_image);
-  add_PL_button = (GtkWidget*) gtk_tool_button_new (tmp_image, _("Playlist"));
+  tmp_toolbar_icon = gtk_image_new_from_stock ("gtk-add", gtk_toolbar_get_icon_size (GTK_TOOLBAR (toolbar)));
+  add_PL_button = gtk_toolbar_append_element (GTK_TOOLBAR (toolbar),
+                                GTK_TOOLBAR_CHILD_BUTTON,
+                                NULL,
+                                _("Playlist"),
+                                _("Add playlist from file"), NULL,
+                                tmp_toolbar_icon, NULL, NULL);
+  gtk_label_set_use_underline (GTK_LABEL (((GtkToolbarChild*) (g_list_last (GTK_TOOLBAR (toolbar)->children)->data))->label), TRUE);
   gtk_widget_show (add_PL_button);
-  gtk_container_add (GTK_CONTAINER (toolbar), add_PL_button);
-  gtk_tool_item_set_tooltip (GTK_TOOL_ITEM (add_PL_button), tooltips, _("Add playlist from file"), NULL);
 
-  tmp_image = gtk_image_new_from_stock ("gtk-justify-left", tmp_toolbar_icon_size);
-  gtk_widget_show (tmp_image);
-  new_PL_button = (GtkWidget*) gtk_tool_button_new (tmp_image, _("New PL"));
+  tmp_toolbar_icon = gtk_image_new_from_stock ("gtk-justify-left", gtk_toolbar_get_icon_size (GTK_TOOLBAR (toolbar)));
+  new_PL_button = gtk_toolbar_append_element (GTK_TOOLBAR (toolbar),
+                                GTK_TOOLBAR_CHILD_BUTTON,
+                                NULL,
+                                _("New PL"),
+                                _("Create New Playlist"), NULL,
+                                tmp_toolbar_icon, NULL, NULL);
+  gtk_label_set_use_underline (GTK_LABEL (((GtkToolbarChild*) (g_list_last (GTK_TOOLBAR (toolbar)->children)->data))->label), TRUE);
   gtk_widget_show (new_PL_button);
-  gtk_container_add (GTK_CONTAINER (toolbar), new_PL_button);
-  gtk_tool_item_set_tooltip (GTK_TOOL_ITEM (new_PL_button), tooltips, _("Create New Playlist"), NULL);
 
-  tmp_image = gtk_image_new_from_stock ("gtk-goto-bottom", tmp_toolbar_icon_size);
-  gtk_widget_show (tmp_image);
-  export_button = (GtkWidget*) gtk_tool_button_new (tmp_image, _("Sync"));
+  tmp_toolbar_icon = gtk_image_new_from_stock ("gtk-goto-bottom", gtk_toolbar_get_icon_size (GTK_TOOLBAR (toolbar)));
+  export_button = gtk_toolbar_append_element (GTK_TOOLBAR (toolbar),
+                                GTK_TOOLBAR_CHILD_BUTTON,
+                                NULL,
+                                _("Sync"),
+                                _("Write Changes to Disk/iPod"), NULL,
+                                tmp_toolbar_icon, NULL, NULL);
+  gtk_label_set_use_underline (GTK_LABEL (((GtkToolbarChild*) (g_list_last (GTK_TOOLBAR (toolbar)->children)->data))->label), TRUE);
   gtk_widget_show (export_button);
-  gtk_container_add (GTK_CONTAINER (toolbar), export_button);
-  gtk_tool_item_set_tooltip (GTK_TOOL_ITEM (export_button), tooltips, _("Write Changes to Disk/iPod"), NULL);
 
-  tmp_image = gtk_image_new_from_stock ("gtk-stop", tmp_toolbar_icon_size);
-  gtk_widget_show (tmp_image);
-  stop_button = (GtkWidget*) gtk_tool_button_new (tmp_image, _("Stop"));
-  gtk_container_add (GTK_CONTAINER (toolbar), stop_button);
-  gtk_tool_item_set_tooltip (GTK_TOOL_ITEM (stop_button), tooltips, _("Stop Display Update"), NULL);
+  tmp_toolbar_icon = gtk_image_new_from_stock ("gtk-stop", gtk_toolbar_get_icon_size (GTK_TOOLBAR (toolbar)));
+  stop_button = gtk_toolbar_append_element (GTK_TOOLBAR (toolbar),
+                                GTK_TOOLBAR_CHILD_BUTTON,
+                                NULL,
+                                _("Stop"),
+                                _("Stop Display Update"), NULL,
+                                tmp_toolbar_icon, NULL, NULL);
+  gtk_label_set_use_underline (GTK_LABEL (((GtkToolbarChild*) (g_list_last (GTK_TOOLBAR (toolbar)->children)->data))->label), TRUE);
 
   paned0 = gtk_hpaned_new ();
   gtk_widget_show (paned0);
